pkg/patterns/declarative/reconciler.go

```diff
@@ -493,9 +493,12 @@ func (r *Reconciler) injectOwnerRef(ctx context.Context, instance DeclarativeObj
 			continue
 		}
 
-		// TODO, error/skip if:
-		// - owner is namespaced and o is not
-		// - owner is in a different namespace than o
+		if owner.GetNamespace() != "" && owner.GetNamespace() != o.Namespace {
+			// a namespaced object can only own objects within the same namespace, not objects in other namespaces or cluster-scoped objects
+			// for any other combination, skip setting owner reference here, to allow declarative.SourceAsOwner to be used for the
+			// subset of objects that make up a supported combination
```

Comment:
This condition should be logged like above.

Reply:
I agree we should alert, but I've been thinking about who we are alerting for.  I think the answer is that we are alerting for the addon operator developer, but not the addon operator user.  There's actually a whole set of these, particularly as we improve the functionality (e.g. reuse of the RESTMapper), which include recommended but not required code changes.

I think I'm going to merge this PR as-is therefore, but then try to implement a "developer notification" mode.

Reply:
@justinsb I'm happy to amend this PR with logging if you'd like, but we currently have a rather ugly workaround with a custom `declarative.WithObjectTransform` that would be nice to get rid of, so if merging this and then adding new logging mechanism can be done in separate step's I'll be happy about it :)